File: src/jp/mcbe/fuyutsuki/Texter/task/SendTextsTask.php
<?php

declare(strict_types=1);

namespace jp\mcbe\fuyutsuki\Texter\task;

use jp\mcbe\fuyutsuki\Texter\data\FloatingTextData;
use jp\mcbe\fuyutsuki\Texter\text\FloatingTextCluster;
use jp\mcbe\fuyutsuki\Texter\text\SendType;
use pocketmine\player\Player;
use pocketmine\scheduler\Task;
use pocketmine\world\World;
use function array_shift;

class SendTextsTask extends Task {

	public const DELAY_TICKS = 5;// 0.25s
	public const TICKING_PERIOD = 2;// 0.1s

	/** @var FloatingTextCluster[] */
	private array $remain;

	public function __construct(
		private readonly Player $target,
		World                   $sendTo,
		private readonly SendType $type
	) {
		$data = FloatingTextData::getInstance($sendTo->getFolderName());
		$this->remain = $data !== null ? $data->floatingTexts() : [];
	}

	public function onRun(): void {
		if (empty($this->remain) || !$this->target->isConnected()) {
			$this->onSuccess();
		}else {
			$floatingText = array_shift($this->remain);
			$floatingText->sendToPlayer($this->target, $this->type);
		}
	}

	private function onSuccess(): void {
		$this->getHandler()->cancel();
	}
}

File: src/main/java/jp/mcbe/fuyutsuki/Texter/task/SendTextsTask.java
package jp.mcbe.fuyutsuki.Texter.task;

import jp.mcbe.fuyutsuki.Texter.data.FloatingTextData;
import jp.mcbe.fuyutsuki.Texter.text.FloatingTextCluster;
import jp.mcbe.fuyutsuki.Texter.text.SendType;
import org.powernukkitx.Player;
import org.powernukkitx.level.Level;
import org.powernukkitx.scheduler.Task;

import java.util.ArrayDeque;
import java.util.Deque;

public final class SendTextsTask extends Task {

	public static final int DELAY_TICKS = 5;   // 0.25s
	public static final int TICKING_PERIOD = 2; // 0.1s

	private final Player target;
	private final SendType type;
	private final Deque<FloatingTextCluster> remain;

	public SendTextsTask(Player target, Level sendTo, SendType type) {
		this.target = target;
		this.type = type;

		FloatingTextData data = FloatingTextData.getInstance(sendTo.getFolderName());
		this.remain = data != null
			? new ArrayDeque<>(data.floatingTexts().values())
			: new ArrayDeque<>();
	}

	@Override
	public void onRun(int currentTick) {
		if (remain.isEmpty() || !target.isConnected()) {
			onSuccess();
		} else {
			FloatingTextCluster floatingText = remain.poll();
			assert floatingText != null;
			floatingText.sendToPlayer(target, type);
		}
	}

	private void onSuccess() {
		cancel();
	}
}